src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -605,6 +728,85 @@ public void withLock(boolean waitForRecovery, ChannelConsumer consumer) throws I
             }
         }
 
+        public void retryingPublishWithLock(ChannelConsumer consumer, ConnectionManagerConfiguration configuration) throws InterruptedException {
+            lock.lock();
+            Iterator<RetryingDelay> iterator = configuration.createRetryingDelaySequence().iterator();
+            try {
+                Channel tempChannel = getChannel(true);
+                while (true) {
+                    try {
+                        consumer.consume(tempChannel);
+                        break;
+                    } catch (IOException | ShutdownSignalException e) {
+                        var currentValue = iterator.next();
+                        int recoveryDelay = currentValue.getDelay();
+                        LOGGER.warn("Retrying publishing #{}, waiting for {}ms. Reason: {}", currentValue.getTryNumber(), recoveryDelay, e);
+                        TimeUnit.MILLISECONDS.sleep(recoveryDelay);
+
+                        // we should not recover channel if it's connection is closed, because if we'll do it channel
+                        // will be also auto recovered by RabbitMQ client during recovering of connection, and we'll
+                        // get two new channels instead of one closed.
```

Comment:
```suggestion
                        // We should not recover the channel if its connection is closed
                        // If we do that the channel will be also auto recovered by RabbitMQ client
                        // during connection recovery, and we will get two new channels instead of one closed.
```